receiver/vcenterreceiver/metrics.go

```diff
@@ -140,6 +141,23 @@ func (v *vcenterMetricScraper) recordVMStats(
 		return
 	}
 	v.mb.RecordVcenterVMCPUUtilizationDataPoint(ts, 100*float64(cpuUsage)/float64(cpuLimit))
+
+	// OverallCpuReadiness is only available in vSphere API 7.0
+	// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/d1902b0e-d479-46bf-8ac9-cee0e31e8ec0/07ce8dbd-db48-4261-9b8f-c6d3ad8ba472/vim.vm.Summary.QuickStats.html
+	vsphereAPIVersion, err := version.NewVersion(v.scrapeData.apiVersion)
```

Comment:
I went back and forth on whether I was going to comment more on this or just leave it be.

It seems like this logic could be moved to a new function (or maybe even to the client as a helper function). The client could be reworked to have a private method like "refreshVsphereAPIVersion" which stores its own version. We could make sure we call this once a collection (or on client initialization). Then we could have another method `vsphereAPIVersionMeetsMin(minVsphereAPIVersion string) bool` that would be used here where you just pass in "7.0.0". It could do all this logic (making use of the stored version (possibly calling `refresh` if it's empty) and then return a true/false for the conditional for lines 158/159. This would be reusable for other cases too.

Ultimately I think I'm fine with it as is, just thinking that it feels a little out of place here and would likely be better placed elsewhere.

Reply:
This is probably a good idea, there may be more of these metrics that require a check on the API in the future. I'll look into changing it up!

Reply:
@jlg-io That JSON API is a totally different (and would require major changes to the vcenterreceiver) from how we're currently collecting. In addition, it looks like it's only been available since vCenter 8.0+. I do agree that a note in the README would be a good add.

Reply:
@StefanKurek, my bad! I guess the right one is there: https://dp-downloads.broadcom.com/api-content/apis/API_VMA_001/8.0U2/html/vim.vm.Summary.QuickStats.html

Reply:
Thanks! I'll add in a section in the readme and also update the link in the comments.